operator/pkg/gateway-api/cell.go

```diff
@@ -76,8 +76,7 @@ func initGatewayAPIController(params gatewayAPIParams) error {
 
 	params.Logger.WithField("requiredGVK", requiredGVK).Info("Checking for required GatewayAPI resources")
 	if err := checkRequiredCRDs(context.Background(), params.K8sClient); err != nil {
-		params.Logger.WithError(err).Error("Required GatewayAPI resources are not found, please refer to docs for installation instructions")
-		return nil
+		return fmt.Errorf("required GatewayAPI resources are not found, please refer to docs for installation instructions: %w", err)
```

Comment:
I am on the fence on this, as returning the error here will crash the operator :thinking:

Reply:
That's exactly the intention of this PR :) It's a missconfiguration and should be made visible the same way as we do for other use-cases - by letting the operator crash. It's not something the operator can recover from once the CRDs are installed afterwards.

IMO: Otherwise it's just hidden somewhere in the logs and the user wonders why it's not working as expected.

Reply:
In addition to this PR it would also be possible to add the check to the Helm Chart. This way we would prevent the user from installing it in the first place.